### PR TITLE
Prevent log warnings and redscreens when equipping Alien Hunters armor in the Shell

### LINKED ISSUE #1090: Add 'none' checks for XComHQ to X2Item_DLC_Day60Armors
Similarly to #1086, equipping the Alien Hunters items outside an actual campaign causes redscreens when the functions like `SetIcarusSuitArmorTint()` fail to access XComHQ.

### PR DESCRIPTION
**Pull request by Iridar**
Fixes #1090 

Wasn't really sure how to classify this issue, so I put inline docs under the "bugfix".

### CODE REVIEW

**Review by Xymanek — COMMENTED** [review]
Technically there is no need to `RequestGameArchetype(ArmorTemplate.EquipNarrative)` if `XComHQ == none`, since the condition later will never succeed, but it doesn't hurt so I'm fine if you want to leave it as-is

**Comment by Iridar** [comment]
> Technically there is no need to `RequestGameArchetype(ArmorTemplate.EquipNarrative)` if `XComHQ == none`, since the condition later will never succeed, but it doesn't hurt so I'm fine if you want to leave it as-is

Nah you make a good point, I'll adjust.

**Comment by Iridar** [comment]
Done so.

**Comment by Xymanek** [comment]
Better compare: https://github.com/X2CommunityCore/X2WOTCCommunityHighlander/compare/8b698dc17efc7e3aaeabca1cf9d52d546222653b...1f9410ecc4a8267b1730cfe5189c8a9c185e07d7

File: Components/DLC2CommunityHighlander/DLC2CommunityHighlander/Src/DLC_2/Classes/X2Item_DLC_Day60Armors.uc [hunk 1]
```diff
@@ -0,0 +1,262 @@
...
+	local X2ArmorTemplate ArmorTemplate;
+	local XComNarrativeMoment EquipNarrativeMoment;
+
+	ArmorTemplate = X2ArmorTemplate(ItemState.GetMyTemplate());
+
+	if (ArmorTemplate != none)
+	{
+		XComHQ = XComGameState_HeadquartersXCom(`XCOMHISTORY.GetSingleGameStateObjectForClass(class'XComGameState_HeadquartersXCom', true)); // Issue #1090 - Allow XComHQ to be 'none'
+		EquipNarrativeMoment = XComNarrativeMoment(`CONTENT.RequestGameArchetype(ArmorTemplate.EquipNarrative));
+		if (XComHQ != none) // Issue #1090 - 'none' check XComHQ before accessing it.
```
Comment (Xymanek) [inline Components/DLC2CommunityHighlander/DLC2CommunityHighlander/Src/DLC_2/Classes/X2Item_DLC_Day60Armors.uc:246]: Any reason why this `RequestGameArchetype()` call wasn't moved inside the `if (XComHQ != none)` check?

Comment (Iridar) [reply Components/DLC2CommunityHighlander/DLC2CommunityHighlander/Src/DLC_2/Classes/X2Item_DLC_Day60Armors.uc]: > Any reason why this `RequestGameArchetype()` call wasn't moved inside the `if (XComHQ != none)` check?

Just an oversight, thanks for noticing. Should be fixed now.

**Comment by Xymanek** [comment]
Looks good, but I can't squash merge this PR since the first commit adds a new file. Can you please force push this PR, squashing the commits with your changes (last 3) into a single one

**Comment by Iridar** [comment]
> Looks good, but I can't squash merge this PR since the first commit adds a new file. Can you please force push this PR, squashing the commits with your changes (last 3) into a single one

Ah, of course, didn't think about that. Was assuming a squash merge is coming, that's why last commit message was so placeholder-y.

**Pull request merged**